### ISSUE TITLE
An unhandled exception of type 'System.StackOverflowException' occurred in Microsoft.Extensions.Configuration.Json.dll

### ISSUE THREAD

**Issue by bhaidar** [aspnet/Home#2257]
Hello,
I am facing an exception when trying to build the ConfigurationBuilder.
An exception is thrown here:
````
return builder.Build(); 
````

The code is shown below. How can I debug such a case?

ASP.NET Core 2

The code I have is something like that:
````
var builder = new ConfigurationBuilder()
                .SetBasePath(path)
                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true);

if (!string.IsNullOrWhiteSpace(environmentName))
{
     builder = builder.AddJsonFile($"appsettings.{environmentName}.json", optional: true);
}

builder = builder.AddEnvironmentVariables();
return builder.Build();            
````

Thanks

**Comment by Tratcher** [comment]
What's the stack trace of the exception?

**Comment by bhaidar** [comment]
Hi @Tratcher 
I am not able to get anything. That's all that I am getting when I run the application.

**Comment by davidfowl** [comment]
Can you provide a repro project on GitHub that shows the problem so that we can run it?

**Comment by bhaidar** [comment]
@davidfowl Can I send it by email? Thanks

**Comment by Tratcher** [comment]
Sure, check my profile.

**Comment by prashanth2** [comment]
I was able to get around this issue by installing NewtonSoft.Json from the package manager console. 

It's not entirely clear how this addresses the issue - but I basically utilized a simplified version of the initialization code in CreateDefaultBuilder and removed the following lines as the exception appeared to be related to loading configurations from appsettings.json:

``config.AddJsonFile("appsettings.json", true, true).AddJsonFile(string.Format("appsettings.{0}.json", (object) hostingEnvironment.EnvironmentName), true, true);``

When I restarted the app, it then failed on the following line but with a clear message indicating that the project requires NewtonSoft.Json.

``services.AddMvc();``

Once I added NewtonSoft.Json and went back to the original CreateDefaultBuilder, the app started up without hiccups. Perhaps there was an issue with the restore process and didn't bring in NewtonSoft.Json by default.
